File: Server/TravelJournal/src/main/java/TravelJournal/payload/request/SignupRequest.java
package TravelJournal.payload.request;

import lombok.Getter;
import lombok.Setter;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import java.util.Set;

/**
 * Parameters necessary for user to create account.
 */
@Getter
@Setter
public class SignupRequest {
    @NotBlank
    private String login;

    @NotBlank
    @Email
    private String email;


    private Set<String> roles;

    @NotBlank
    private String password;
}

File: Client/app/src/main/java/com/example/traveljournal/data/model/request/SignUpRequest.kt
package com.example.traveljournal.data.model.request

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class SignUpRequest {
    @SerializedName("login")
    @Expose
    var login: String? = null

    @SerializedName("email")
    @Expose
    var email:String? = null

    @SerializedName("password")
    @Expose
    var password: String? = null
}